packages/flutter_tools/test/general.shard/test/test_compiler_test.dart

```diff
@@ -129,12 +106,46 @@ class FakeTestCompiler extends TestCompiler {
     this.residentCompiler,
   ) : super(buildInfo, flutterProject);
 
-  final MockResidentCompiler residentCompiler;
+  final FakeResidentCompiler residentCompiler;
 
   @override
   Future<ResidentCompiler> createCompiler() async {
     return residentCompiler;
   }
 }
 
-class MockResidentCompiler extends Mock implements ResidentCompiler {}
+class FakeResidentCompiler extends Fake implements ResidentCompiler {
```

Comment:
What does extending Fake give you?

Reply:
I didn't implement reset since it isn't used